src/game_battlealgorithm.cpp

```diff
@@ -925,12 +926,24 @@ bool Game_BattleAlgorithm::Normal::Execute() {
 		// FIXME: Differentiate the cases where 2k3 back attack bug applies
 		auto effect = Algo::CalcNormalAttackEffect(source, target, Game_Battler::WeaponAll, critical_hit, true, Game_Battle::GetBattleCondition(), true);
 		effect = Algo::AdjustDamageForDefend(effect, target);
-		// FIXME: Handle negative effect from attributes
+
+		// Handle negative effect from attributes
+		if (effect < 0) {
+			this->negative_effect = true;
+			this->healing = true;
+			effect = -effect;
+		}
+
 		effect = Utils::Clamp(effect, 0, MaxDamageValue());
 
 		this->hp = effect;
 
-		if (GetTarget()->GetHp() - this->hp <= 0) {
+		// RPG_RT bug: Negative effects affect HP double
+		if (this->negative_effect) {
+			this->hp *= 2;
```

Comment:
One more edge case, the double effect bug *does not* happen for absorb hp. Only normal hp damage..